sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubListener.cs

```diff
@@ -190,14 +302,52 @@ public async Task ProcessEventsAsync(EventProcessorHostPartition context, IEnume
                     else
                     {
                         // Batch dispatch
-                        TriggeredFunctionData input = new TriggeredFunctionData
+
+                        if (_minimumBatchesEnabled)
                         {
-                            TriggerValue = triggerInput,
-                            TriggerDetails = triggerInput.GetTriggerDetails(context)
-                        };
+                            var triggerEvents = StoredEventsManager.ProcessWithStoredEvents(context, messages.ToList(), false, linkedCts.Token);
+
+                            if (triggerEvents.Length > 0)
+                            {
+                                UpdateCheckpointContext(events, context);
+                                await TriggerExecute(triggerEvents, context, linkedCts.Token).ConfigureAwait(false);
+                                eventToCheckpoint = triggerEvents.Last();
+
+                                if (_storedEventsBackgroundTask != null)
+                                {
+                                    // If there is a background timer task, cancel it and dispose of the cancellation token.
+                                    _storedEventsBackgroundTaskCts.Cancel();
+                                    _storedEventsBackgroundTaskCts.Dispose();
+                                    _storedEventsBackgroundTaskCts = null;
+                                }
+
+                                if (StoredEventsManager.HasStoredEvents)
```

Comment:
I think we want to treat each partition independently.   Trying to keep them all on the same flush schedule while reads are potentially inconsistent doesn't seem the right thing to do.   Am I overlooking something?

Reply:
Ignore me... wrong scope.